tcms/issuetracker/tests/test_azure.py

```diff
@@ -0,0 +1,170 @@
+        initial_comments_length = initial_comments['count']
+        current_comments_length = initial_comments_length
+        while current_comments_length != initial_comments_length + 1:
+            comments = self.integration.rpc.get_comments(self.existing_bug_id)
+            current_comments_length = comments['count']
+
+        last_comment = comments['comments'][0]
+
+        # assert that a comment has been added as the last one
+        # and also verify its text
+        for expected_string in [
+                'Confirmed via test execution',
+                "TR-%d: %s" % (self.execution_1.run_id, self.execution_1.run.summary),
+                self.execution_1.run.get_full_url(),
+                "TE-%d: %s" % (self.execution_1.pk, self.execution_1.case.summary)]:
+            self.assertIn(expected_string, last_comment['text'])
+
+    def test_report_issue_from_test_execution_1click_works(self):
+        # simulate user clicking the 'Report bug' button in TE widget, TR page
+        result = self.rpc_client.Bug.report(self.execution_1.pk, self.integration.bug_system.pk)
+        self.assertEqual(result['rc'], 0)
+        self.assertIn(self.integration.bug_system.base_url, result['response'])
+        self.assertIn('/_workitems/edit/', result['response'])
+
+        new_issue_id = self.integration.bug_id_from_url(result['response'])
+        issue = self.integration.rpc.get_issue(new_issue_id)
+
+        self.assertEqual("Failed test: %s" % self.execution_1.case.summary,
+                         issue['fields']['System.Title'])
+        for expected_string in [
+                "Filed from execution %s" % self.execution_1.get_full_url(),
+                self.execution_1.run.plan.product.name,
+                self.component.name,
+                "Steps to reproduce",
+                self.execution_1.case.text]:
+            self.assertIn(expected_string, issue['fields']['System.Description'])
+
+        # verify that LR has been added to TE
+        self.assertTrue(LinkReference.objects.filter(
+            execution=self.execution_1,
+            url=result['response'],
+            is_defect=True,
+        ).exists())
+
+    def test_report_issue_from_test_execution_empty_baseurl_exception(self):
+        # simulate user clicking the 'Report bug' button in TE widget, TR page
+        bug_system = BugSystem.objects.create(  # nosec:B106:hardcoded_password_funcarg
+            name='Azure for kiwitcms/test-azure-integration Exception',
+            tracker_type='tcms.issuetracker.azure_boards.AzureBoards',
+            base_url='incorrect_url',
+            api_password=os.getenv('AZURE_BOARDS_INTEGRATION_API_TOKEN'),
+        )
+        integration = AzureBoards(bug_system, None)
+        result = self.rpc_client.Bug.report(self.execution_1.pk, integration.bug_system.pk)
+        self.assertIn('_workitems/create/Issue', result['response'])
+
+
+class TestAzureAPI(APITestCase):
```

Comment:
missing skipUnless decorator

Reply:
Can also inherit from the standard unittest.TestCase class b/c we don't need anything Django specific here.